src/cuga/backend/tools/image_analysis.py

```
@@ -0,0 +1,217 @@
+
+def _build_data_url(source: str) -> str:
+    """Return a ``data:<media>; base64,...`` URL from a local path or HTTP URL."""
+    if _is_url(source):
+        import urllib.request
+
+        suffix = Path(source.split("?")[0]).suffix or ".jpg"
+        dest = Path(f"_img_download{suffix}")
+        req = urllib.request.Request(
+            source,
+            headers={"User-Agent": "Mozilla/5.0 (compatible; CugaAgent/1.0)"},
+        )
```

Ruff (0.15.20)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)